{bio}[foss/2023a] MAGeCK-VISPR v0.5.6

grpiccoli opened the pull request:
(created using `eb --new-pr`)

Micket commented on easybuild/easyconfigs/m/MAGeCK-VISPR/MAGeCK-VISPR-0.5.6-foss-2023a.eb:
```diff
@@ -0,0 +1,153 @@
+# #
+# This is a contribution from HPCNow! (http://hpcnow.com)
+# Copyright::   HPCNow!
+# Authors::     Jordi Blasco <jordi.blasco@hpcnow.com>; Guillermo AR Piccoli <guillermo.piccoli@hpcnow.com>
+# License::     GPL-v3.0
+# #
```
huh?

grpiccoli replied:
Is that header alright? should I include it? These are all updated unsubmitted recipes that were developed internally by HPCNow! before, that's why Jordi is there,
Cheers

Micket commented on easybuild/easyconfigs/m/MAGeCK-VISPR/MAGeCK-VISPR-0.5.6-foss-2023a.eb:
```diff
@@ -0,0 +1,153 @@
+# License::     GPL-v3.0
+# #
+
+easyblock = 'PythonPackage'
```
Should be a PythonBundle

Micket commented on easybuild/easyconfigs/m/MAGeCK-VISPR/MAGeCK-VISPR-0.5.6-foss-2023a.eb:
```diff
@@ -0,0 +1,153 @@
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+
+source_urls = ['https://bitbucket.org/liulab/%(namelower)s/get']
+sources = ['v%(version)s.tar.gz']
+checksums = ['b6bdd59d34f9bbb548c88de8948e2aacf7e461a62a74c1b8ef8a73b9b64b2e92']
```
should be one of the extensions in the bundle instead

Micket commented on easybuild/easyconfigs/m/MAGeCK-VISPR/MAGeCK-VISPR-0.5.6-foss-2023a.eb:
```diff
@@ -0,0 +1,153 @@
+
+dependencies = [
+    ('Python', '3.11.3'),
+    ('Meson', '1.2.3', '', ('GCCcore', '13.2.0'))
```
wrong toolchain, don't specify the toolchain at all, and Meson should just be a builddep

grpiccoli replied:
This requires Meson >= 1.2.1 but it is not available in GCC 12.3.0, should I commit a new Meson eb for 12.3.0 or should I just switch MAGeCK-VISPR to foss 2023b?

Micket replied:
You can switch to 2023b if you want.

Adding multiple versions of things that are builddependencies is completely OK; they are only used during the build and won't cause conflicts afterwards (which is the challenge in case of things used as normal deps). So your other option is to make a new `Meson-1.4.0-GCCcore-12.3.0.eb`

grpiccoli replied:
Thank you @Micket I pr Meson-1.4.0-GCCcore-12.3.0.eb as per your advice, because some functions in MAGeCK-VISPR are obsolete in Python 3.11.5 so it is incompatible with the available Meson 1.2 in EB.

Micket replied:
Really? Something was deprecated in 3.11.5 from 3.11.3 which we are using here? Surely not as I would hope python follows semantic versioning. It's fine to keep this in 12.3.0 if you prefer this toolchain, but I don't see how Python makes any difference here.

Micket commented on easybuild/easyconfigs/m/MAGeCK-VISPR/MAGeCK-VISPR-0.5.6-foss-2023a.eb:
```diff
@@ -0,0 +1,153 @@
+use_pip = True
+sanity_pip_check = True
+
+options = {'modulename': 'mageck_vispr'}
+
+exts_defaultclass = 'PythonPackage'
+
+exts_default_options = {
+    'download_dep_fail': True,
+    'sanity_pip_check': True,
+    'source_urls': [PYPI_SOURCE],
+    'use_pip': True,
+}
```
```suggestion
```

Micket commented on easybuild/easyconfigs/m/MAGeCK-VISPR/MAGeCK-VISPR-0.5.6-foss-2023a.eb:
```diff
@@ -0,0 +1,153 @@
+    'use_pip': True,
+}
+
+exts_list = [
```
Almost all (maybe all?) of the extensions are part of Python bundles we have already. In particular Python-bundle-PyPI and SciPy-bundle. The hatchling + packaging stuff should be added as builddeps.

Micket commented on easybuild/easyconfigs/m/MAGeCK-VISPR/MAGeCK-VISPR-0.5.6-foss-2023a.eb:
```diff
@@ -0,0 +1,74 @@
+
+dependencies = [
+    ('SciPy-bundle', '2023.07'),
+    ('Meson', '1.4.0')
```
Meson would be a build dependency.

Micket commented on easybuild/easyconfigs/m/MAGeCK-VISPR/MAGeCK-VISPR-0.5.6-foss-2023a.eb:
```diff
@@ -0,0 +1,74 @@
+sanity_pip_check = True
+
+exts_list = [
+    ('PyYAML', '5.4.1', {
```
We have a PyYAML module

Micket commented on easybuild/easyconfigs/m/MAGeCK-VISPR/MAGeCK-VISPR-0.5.6-foss-2023a.eb:
```diff
@@ -0,0 +1,74 @@
+    ('calver', '2022.6.26', {
+        'checksums': ['e05493a3b17517ef1748fbe610da11f10485faa7c416b9d33fd4a52d74894f8b'],
+    }),
+    ('trove_classifiers', '2024.3.3', {
+        'source_tmpl': 'trove-classifiers-%(version)s.tar.gz',
+        'checksums': ['df7edff9c67ff86b733628998330b180e81d125b1e096536d83ac0fd79673fdc'],
+    }),
+    ('hatchling', '1.22.4', {
+        'checksums': ['8a2dcec96d7fb848382ef5848e5ac43fdae641f35a08a3fab5116bd495f3416e'],
+    }),
+    ('hatch-vcs', '0.4.0', {
+        'source_tmpl': 'hatch_vcs-%(version)s.tar.gz',
+        'checksums': ['093810748fe01db0d451fabcf2c1ac2688caefd232d4ede967090b1c1b07d9f7'],
+    }),
+    ('hatch-fancy-pypi-readme', '24.1.0', {
+        'source_tmpl': 'hatch_fancy_pypi_readme-%(version)s.tar.gz',
+        'checksums': ['44dd239f1a779b9dcf8ebc9401a611fd7f7e3e14578dcf22c265dfaf7c1514b8'],
+    }),
```
Replace with `('hatchling', '1.18.0'),`  (build dependency only)

Micket commented on easybuild/easyconfigs/m/MAGeCK-VISPR/MAGeCK-VISPR-0.5.6-foss-2023a.eb:
```diff
@@ -0,0 +1,74 @@
+        'modulename': 'yaml',
+        'checksums': ['607774cbba28732bfa802b54baa7484215f530991055bb562efbed5b2f20a45e'],
+    }),
+    ('pyproject-metadata', '0.7.1', {
+        'checksums': ['0a94f18b108b9b21f3a26a3d541f056c34edcb17dc872a144a15618fed7aef67'],
+    }),
```
this looks like something intended to be picked up as a dependency for meson-python, so, i would say just use meson-python-0.15.0-GCCcore-12.3.0.eb. Seeing this, I now wonder if this was the only reason for updating Meson anyway.. which means that adding it wasn't actually needed at all and it just stems from this.

Micket commented on easybuild/easyconfigs/m/MAGeCK-VISPR/MAGeCK-VISPR-0.5.6-foss-2023a.eb:
```diff
@@ -0,0 +1,74 @@
+    ('pyproject-metadata', '0.7.1', {
+        'checksums': ['0a94f18b108b9b21f3a26a3d541f056c34edcb17dc872a144a15618fed7aef67'],
+    }),
+    ('setuptools-scm', '8.0.4', {
+        'modulename': 'setuptools_scm',
+        'checksums': ['b5f43ff6800669595193fd09891564ee9d1d7dcb196cab4b2506d53a2e1c95c7'],
+    }),
+    ('calver', '2022.6.26', {
+        'checksums': ['e05493a3b17517ef1748fbe610da11f10485faa7c416b9d33fd4a52d74894f8b'],
+    }),
```
I struggle to see where this requirements come from.. In fact. there doesn't seem to be much of *any* overlap between all these extensions. 
MAGeCK-VISPR lists 
fastqc >=0.11.4
mageck >=0.5.9
vispr >=0.4.11
snakemake >=3.6.0
cutadapt >=1.9.1
jinja2 >=2.8
which really has nothing in common with anything here. 

In addition, this software has basically been abandoned for some 3 years now, so the idea that it would have requirements on newer versions of anything does not seem reasonable to me. 


Edit: I just tested this now, removed Meson, removed all these extensions, added PyYAML to the dependencies, and it builds fine like this.
I looked through the source code and it needs
1. pandas (provided by SciPy-bundle)
2. jinja2 (also happens to be provided by SciPy-bundle)
3. yaml (provided by PyYaml)
No other extensions are needed here at all.

Micket commented on easybuild/easyconfigs/m/MAGeCK-VISPR/MAGeCK-VISPR-0.5.6-foss-2023a.eb:
```diff
@@ -0,0 +1,74 @@
+    (name, version, {
+        'source_tmpl': 'v%(version)s.tar.gz',
+        'modulename': 'mageck_vispr',
+        'source_urls': 'https://bitbucket.org/liulab/%(namelower)s/get',
```
```suggestion
        'source_urls': ['https://bitbucket.org/liulab/%(namelower)s/get'],
```

grpiccoli commented:
@Micket thanks again, I removed all dependencies and the Meson eb file as suggested and used meson-python, hatchling, and PyYAML instead and changed it back to PythonPackage because it is no longer a Bundle.
Cheers!
